### PR TITLE
fix #3371 feat(visualization) Add secondary metric data and component.

### LINKED ISSUE #3371: Add secondary metric data tables
┆Issue is synchronized with this [Jira Task](https://jira.mozilla.com/browse/EXP-249)
┆Issue Number: EXP-249

### PR DESCRIPTION
**Pull request by emtwo**

### CODE REVIEW

**Comment by emtwo** [comment]
I didn't add this into ExperimentDetails since secondary metrics are not available there yet. We'll do that in nimbus ui once this moves out of legacy

File: app/experimenter/legacy-ui/rapid/__tests__/secondaryMetricTable.test.tsx [hunk 1]
```diff
@@ -0,0 +1,89 @@
+import { render, act, screen } from "@testing-library/react";
+import React from "react";
+
+import { EXPERIMENT_DATA } from "experimenter-rapid/__tests__/visualizationTestData";
+import SecondaryMetricTable from "experimenter-rapid/components/visualization/SecondaryMetricTable";
+
+const PROBESET_D = "probeset_d";
+
+describe("<SecondaryMetricTable />", () => {
```
Comment (jaredlockhart) [inline app/experimenter/legacy-ui/rapid/__tests__/secondaryMetricTable.test.tsx:9]: I love the way you write js tests, so succinct 💯 💯 💯

**Review by jaredlockhart — COMMENTED** [review]
@emtwo Can you do me a favour and put some screenshots in the PR of all the UI changes?  Makes it easier to evaluate.

File: app/experimenter/visualization/tests/api/test_views.py [hunk 2]
```diff
@@ -206,9 +206,13 @@ def read(self):
 
         mock_open.return_value = File()
         mock_exists.return_value = True
-        probe_set = NimbusProbeSetFactory.create()
+        probe_set_primary = NimbusProbeSetFactory.create()
         experiment = NimbusExperimentFactory.create_with_status(
-            target_status=status, probe_sets=[probe_set]
+            target_status=status, probe_sets=[probe_set_primary]
```
Comment (jaredlockhart) [inline app/experimenter/visualization/tests/api/test_views.py:211]: You might wanna set up some primary and secondary probesets here to catch both cases above?

**Comment by jaredlockhart** [comment]
Also you gotta rebase this on main to get the updated circle config

File: app/experimenter/visualization/tests/api/test_views.py (hunk 2 above)
Comment (emtwo) [reply app/experimenter/visualization/tests/api/test_views.py:211]: The next line below this appends secondary probe sets:
`experiment.probe_sets.add(...)`

Unless there is better way of doing this?

Comment (jaredlockhart) [reply app/experimenter/visualization/tests/api/test_views.py:211]: Oh yes, no, that's right, I didn't read that correctly.  We should probably add some helper methods to the factory for adding primary/secondary probesets, but yes nevermind me that's right 👍

**Review by jaredlockhart — APPROVED** [review]
👍 👍 👍

**Comment by emtwo** [comment]
Screenshot here for reference:

Screen Shot 2020-11-06 at 10 49 33 AM

I didn't add the screenshot initially since I didn't actually add these components to experimentDetails (as explained above)

So when we do add them, they'll have a real title unlike the underscored slug names there. But hopefully the screenshot is enough as a reference.

**Pull request merged**